master/pkg/searcher/search_method.go

```diff
@@ -22,51 +22,52 @@ type SearchMethod interface {
 	trialCreated(ctx context, requestID RequestID) ([]Operation, error)
 	// trainCompleted informs the searcher that the training workload initiated by the same searcher
 	// has completed. It returns any new operations as a result of this workload completing.
-	trainCompleted(ctx context, requestID RequestID, message Workload) ([]Operation, error)
+	trainCompleted(ctx context, requestID RequestID, train Train) ([]Operation, error)
```

Comment:
We talked about this offline. The experiment actor is still relayed workload completed messages for the sole purpose of checking if a validation was the best and recording how many units the trial is completing so that the above doesn't kill searcher granularity.